m3/src/main/java/com/uber/m3/tally/m3/M3Reporter.java

```diff
@@ -136,7 +131,7 @@
 
     // This is a synchronization barrier to make sure that reporter
     // is being shutdown only after all of its processor had done so
-    private final CountDownLatch processorsShutdownLatch;
+    private final Semaphore processorsShutdownSema;
```

Comment:
Are there existing tests for the shutdown behavior? e.g. "reporter shutdown waits for all processors to be shutdown"